core/backend-itwin-client/src/imodelhub/BlobDownloader.ts

```diff
@@ -0,0 +1,591 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
```

Comment:
This file should probably be included in the barrel file since it's used in the API of AzureFileHandler.

Reply:
Thanks, i forgot the azure file handler take ConfigData as optional parameter. 
Note here the other constructor parameter are deprecated or not used. But i did not removed them as that would be breaking change.

Reply:
The following will need to be removed as well along with AzureFileHandler unused constructor variables.
* BufferedStream
* downloadFileAtomic

I tried removing but it used by some test/api (UrlFileHandler.ts) in imodelhub. It would be better if they remove it.

Reply:
Okay, can you make sure to add the proper deprecation and internal API flags on all of the newly added items? I think they'll all be public by default right now.

Reply:
I did added 'internal' to all exported types. Do i need to add internal to any non-exported types?

Reply:
Nope, you're good. I missed those tags earlier.